pyat/at/physics/revolution.py

```diff
@@ -21,25 +24,34 @@ def get_mcf(ring: Lattice, dp: Optional[float] = 0.0,
         dp:             Momentum deviation. Defaults to :py:obj:`None`
         keep_lattice:   Assume no lattice change since the previous tracking.
           Default: :py:obj:`False`
+        fit_order:      Maximum momentum compaction factor to be fitted.
+          Defaults to :py:obj:`None`
+        n_step:         Number of different calculated momentum deviations to be fitted
+          with a polynomial.
+          Defaults to :py:obj:`None`
 
     Keyword Args:
         DPStep (Optional[float]):       Momentum step size.
           Default: :py:data:`DConstant.DPStep <.DConstant>`
 
     Returns:
-        mcf (float):    Momentum compaction factor :math:`\alpha`
+        mcf (array):    Momentum compaction factor :math:`\alpha`
+          up to the order fit_order
     """
+    if n_step < 2*fit_order:
```

Comment:
Here one should raise at leastan AtWarning instead of just printing it out. However you may as well consider rising an AtError in this case to force sufficient number of steps